source/mbed_trace.c

```diff
@@ -97,6 +97,13 @@ static void mbed_trace_realloc(char **buffer, int *length_ptr, int new_length);
 static void mbed_trace_default_print(const char *str);
 static void mbed_trace_reset_tmp(void);
 
+typedef struct trace_setting_s {
+    /** Group name */
+    char grp[4];
```

Comment:
Note that the 4 character limitation isn't otherwise enforced, so there are longer group names in existing code.

Reply:
Oh, indeed.

I initially used a `const char *` there but it would require that the string passed on the first call to `mbed_trace_set_level` stays alive for the rest of the lifetime of the app, so either mbed_trace takes the ownership of it (and never free it) or it is a statically allocated string. But I thought it'd open the gate to some unsound behaviour if the string gets freed. 

Another solution would be to strdup the string but mean wasting some memory.

Reply:
It could just be a (documented) restriction that it must be available for as long as the process runs. To complement that, you could also add API for removing a configuration though?

Reply:
Hi Tommi, I updated the PR with your feedbacks.